packages/mongo-nav/src/MongoNav.spec.tsx

```diff
@@ -1,68 +1,143 @@
 import React from 'react';
-import { render, cleanup, wait } from '@testing-library/react';
+import { render, cleanup } from '@testing-library/react';
 import '@testing-library/jest-dom/extend-expect';
+import { nullableElement, Queries } from 'packages/lib/src/testHelpers';
+import { dataFixtures } from './data';
 import MongoNav from './MongoNav';
 
-afterAll(cleanup);
-
-describe('packages/MongoNav', () => {
-  const onOrganizationChange = jest.fn();
-  const onProjectChange = jest.fn();
-  const cloudHost = 'https://cloud-dev.mongodb.com';
-
-  const { getByText, getByTestId } = render(
-    <MongoNav
-      activeProduct="realm"
-      mode="dev"
-      onOrganizationChange={onOrganizationChange}
-      onProjectChange={onProjectChange}
-      urls={{
-        orgNav: {
-          accessManager: 'https://cloud.mongodb.com/access-manager-test',
-        },
-      }}
-      hosts={{ cloud: cloudHost }}
-    />,
-  );
-
-  describe('by default', () => {
-    test('it renders the ProjectNav', () => {
-      const projectNav = getByTestId('project-nav');
-      expect(projectNav).toBeInTheDocument();
+// types
+interface ExpectedElements {
+  [key: string]: nullableElement;
+}
+
+describe('packages/mongo-nav', () => {
+  const queries: Queries = {};
+  const expectedElements: ExpectedElements = {};
+
+  const setQueries = ({ queryByTestId }: Queries) => {
+    Object.assign(queries, { queryByTestId });
+    setExpectedElements();
+  };
+
+  const setExpectedElements = () => {
+    const { queryByTestId = () => null } = queries;
+    expectedElements.orgNav = queryByTestId('organization-nav');
+    expectedElements.projectNav = queryByTestId('project-nav');
+    expectedElements.admin = queryByTestId('org-nav-admin-link');
+    expectedElements.userMenu = queryByTestId('user-menu-trigger');
+    expectedElements.billing = queryByTestId('org-nav-billing');
+    expectedElements.activeProject = queryByTestId(
+      'project-select-active-project',
+    );
+  };
+
+  let onOrganizationChange: jest.Mock;
+  let onProjectChange: jest.Mock;
+  let fetchMock: jest.Mock;
+
+  beforeEach(() => {
+    onOrganizationChange = jest.fn();
+    onProjectChange = jest.fn();
+    fetchMock = jest.fn();
+    window.fetch = fetchMock;
```

Comment:
Gotta make sure you restore the original fetch function.

Reply:
Updated!

Reply:
If you go to another test does window.fetch exist?

What I had in mind was:

```jsx
let originalFetch = window.fetch;
window.fetch = fetchMock;

// afterEach
window.fetch = originalFetch
```